libptable/ptable.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Partition table
+ *
+ * Copyright 2020, 2023 Phoenix Systems
+ * Author: Hubert Buczynski, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _LIB_PTABLE_H_
+#define _LIB_PTABLE_H_
+
+#include <stdint.h>
+#include <sys/types.h>
+
+
+/* Structure of partition table
+ *  _________________________________________________________________________
+ * |       28 B      |                 32 B * n                |     4 B     |
+ * |-----------------|-----------------------------------------|-------------|
+ * | ptable_t header | ptable_part_t 0 | ... | ptable_part_t n | magic bytes |
+ *  -------------------------------------------------------------------------
+ * 
+ *  NOTE: data in partition table should be stored in little endian
```

Comment:
Lib does not take care of that - CRC is stored per target endianness.

Reply:
CRC is not the only thing, I see, we parse partition info disregarding possibility of our target being big endian.

Reply:
Not really in scope of this PR, but we should take note - we already have one big endian target.